src/java/org/apache/cassandra/auth/CassandraRoleManager.java

```diff
@@ -164,12 +168,9 @@ public CassandraRoleManager()
 
     public CassandraRoleManager(Map<String, String> parameters)
     {
-        supportedOptions = DatabaseDescriptor.getAuthenticator() instanceof PasswordAuthenticator
```

Comment:
can you please rebase this patch on top of current trunk? This constructor was changed in CEP-55 by conditionally adding `Option.OPTIONS` among supportedOptions if there is some specific role name generator.

I think what should happen here is that `getSupportedRoleOptions` should return Set.of(Option.OPTIONS) but only in case `Guardrails.roleNamePolicy.getGenerator() != NoOpGenerator.INSTANCE` as is currently done in trunk.

Basically, my point is that this decision logic should be delegated to IAuthenticator implementation as you introduced that capability instead having it baked in here.